controllers/provider-alicloud/pkg/controller/controlplane/valuesprovider.go

```diff
@@ -218,6 +219,9 @@ func (vp *valuesProvider) GetConfigChartValues(
 	cp *extensionsv1alpha1.ControlPlane,
 	cluster *extensionscontroller.Cluster,
 ) (map[string]interface{}, error) {
+	if cp.Spec.ProviderConfig == nil {
```

Comment:
Instead  of throwing an error, can you simply default the providerConfig to an empty object if it wasn't provided? The functions that actually use it should validate that the required fields are set/return an error in this case.

Something like

```golang
cpConfig := &apisalicloud.ControlPlaneConfig{}
if cp.Spec.ProviderConfig != nil && cp.Spec.ProviderConfig.Raw != nil {
	if _, _, err := vp.decoder.Decode(cp.Spec.ProviderConfig.Raw, nil, cpConfig); err != nil {
		return nil, errors.Wrapf(err, "could not decode providerConfig of controlplane '%s'", util.ObjectName(cp))
	}
}
```

WDYT?

Reply:
Makes sense, although I need to make sure that this validation happens at a later point in time in each provider.

Reply:
Yes, I think this would be a more valuable PR. In the end this will allow end-users to e.g. not specify the `ControlPlaneConfig` in their `Shoots`.
Today, for certain providers (like AWS, for example), they **have** to specify

```yaml
apiVersion: aws.provider.extensions.gardener.cloud/v1alpha1
kind: ControlPlaneConfig
```

which basically has no content.